src/scitacean/dataset.py

```diff
@@ -396,7 +398,7 @@ def _get_or_add_orig_datablock(self, key: Union[int, str]) -> OrigDatablock:
 
     def make_upload_model(self) -> Union[UploadDerivedDataset, UploadRawDataset]:
```

Comment:
Can `UploadDerivedDataset` and `UploadRawDataset` be `TypeVar` instead?
If the model is `Type[UploadRawDataset]`, the returned model is `UploadRawDataset` for example, right...?

Reply:
But this method doesn't take any arguments except `self`. So how would you match the type of the model?

Reply:
Oh Sorry I thought the model was also an input. Never mind...